src/blueprints/tv2/factories/tv2-viz-gfx-timeline-object-factory.ts

```diff
@@ -0,0 +1,114 @@
+import { DeviceType } from '../../../model/enums/device-type'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+
+export class Tv2VizGfxTimelineObjectFactory implements Tv2GfxTimelineObjectFactory {
+  public createThemeOutTimelineObject(blueprintConfiguration: Tv2BlueprintConfiguration, duration: number): VizMseElementInternalTimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.ELEMENT_INTERNAL,
+        templateName: 'OUT_TEMA_H',
+        templateData: [],
+        showName: blueprintConfiguration.showStyle.selectedGfxSetup.OvlShowName ?? ''
+      }
+    }
+  }
+
+  public createOverlayInitializeTimelineObject(duration: number): VizMseLoadAllElementsTimelineObject {
+    return {
+      id: 'loadAllElements',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.LOAD_ALL_ELEMENTS
+      }
+    }
+  }
+
+  public createContinueGfxTimelineObject(duration: number): VizMseContinueTimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: {
+        deviceType: DeviceType.VIZMSE,
+        type: VizType.CONTINUE,
+        direction: 1,
+        reference: Tv2VizLayer.GRAPHIC_PILOT     
+      }
+    }
+  }
+
+  public createGfxClearTimelineObject(blueprintConfiguration: Tv2BlueprintConfiguration, duration: number): VizMseClearGfxTimelineObject {
+    return (this.createBaseTimelineObjectWithContent(
+      duration,
+      this.createGfxClearContent(blueprintConfiguration)
+    ) as VizMseClearGfxTimelineObject) // Todo: figure out if this `as` usage is okay.
+  }
+
+  private createGfxClearContent(blueprintConfiguration: Tv2BlueprintConfiguration): object {
+    return {
+      deviceType: DeviceType.VIZMSE,
+      type: VizType.CLEAR_ALL_ELEMENTS,
+      channelsToSendCommands: ['OVL1', 'FULL1', 'WALL1'],
+      showName: blueprintConfiguration.showStyle.selectedGfxSetup.OvlShowName ?? ''
+    }
+  }
+
+  private createBaseTimelineObjectWithContent(duration: number, content: object): TimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0,
+        duration
+      },
+      priority: 100,
+      layer: Tv2VizLayer.GFX_ACTIONS,
+      content: content
+    }
+  }
+
+  public createGfxAllOutTimelineObject(blueprintConfiguration: Tv2BlueprintConfiguration, duration: number): VizMseClearGfxTimelineObject {
+    return (this.createBaseTimelineObjectWithContent(
+      duration,
+      this.createGfxAllOutContent(blueprintConfiguration)
+    ) as VizMseClearGfxTimelineObject) // Todo: figure out if this `as` usage is okay.
```

Comment:
Same comment as to the other one of this TODO.

Reply:
Same response.